pkg/utils/integration/pipelineruns.go

```diff
@@ -57,6 +57,7 @@ func (i *IntegrationController) CreateIntegrationPipelineRun(snapshotName, names
 }
 
 // GetComponentPipeline returns the pipeline for a given component labels.
+// In case of failure, this function retries till it gets timed out.
```

Comment:
This function has some issue -   pipelineRunLabels are declared, but it's not really being used in the logic.

Reply:
Nice catch! It existed even before my commits, I've fixed it now. Thanks!